### PR TITLE
Add integration for actix_web

### PR DESCRIPTION
**Pull request by ericmarkmartin**
I tried to follow the contributing guidelines best I could but couldn't find anything about what to write for the `Makefile.toml` and `release.toml` files, so I was hoping that could be discussed here. Also, I figured I should wait until merging this before publishing the crate to crates.io, but if there's a different procedure for that as well, please let me know.

Also, with respect to #442, I have already implemented the functionality necessary to accept application/graphql content-typed requests.

### CODE REVIEW

File: docs/book/content/servers/index.md [hunk 1]
```diff
@@ -1,17 +1,10 @@
-# Adding A Server
-
-To allow using Juniper with the HTTP server of your choice,
```
Comment (theduke) [inline docs/book/content/servers/index.md]: Why did you remove this? It remains accurate.

File: juniper_actix_web/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+[dev-dependencies]
+actix-rt = "0.2.0"
+pretty_env_logger = "0.2"
+awc = "0.2.8"
+url = "2.1.0"
+
+[dev-dependencies.juniper]
+version = "0.14.1"
+features = ["expose-test-schema", "serde_json"]
+path = "../juniper"
```
Comment (theduke) [inline juniper_actix_web/Cargo.toml]: Missing a newline. ;)

File: juniper_actix_web/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,620 @@
...
+        Err(de::Error::invalid_length(0, &"a positive integer"))
+    } else {
+        Ok(v)
+    }
+}
+
+#[derive(Debug, serde_derive::Deserialize, PartialEq)]
+#[serde(untagged)]
+#[serde(bound = "InputValue<S>: Deserialize<'de>")]
+enum GraphQLBatchRequest<S = DefaultScalarValue>
```
Comment (theduke) [inline juniper_actix_web/src/lib.rs]: We really need to factor out the batch request into the main crate, right now we are just duplicating it everywhere. 

(You would be very welcome to do this if interested, but just a suggestion. We can do this as a follow up. )

File: juniper_actix_web/src/lib.rs [hunk 4]
```diff
@@ -0,0 +1,620 @@
...
+
+        Ok(HttpResponse::Ok()
+            .status(status)
+            .content_type("application/json")
+            .body(body))
+    }
+}
+
+#[cfg(test)]
+mod fromrequest_tests {
```
Comment (theduke) [inline juniper_actix_web/src/lib.rs]: Since you wrote quite a lot of tests ( :+1: ), let's factor them out to a separate file.

**Comment by theduke** [comment]
> couldn't find anything about what to write for the `Makefile.toml`

We don't need a `Makefile.toml` if no customizations are required. It works automatically.

> I figured I should wait until merging this before publishing the crate to crates.io

If this is a official integration, the crate will have to be owned and published by us.

File: docs/book/content/servers/index.md (hunk 1 above)
Comment (ericmarkmartin) [reply docs/book/content/servers/index.md]: I apologize, I didn't mean to do that

File: juniper_actix_web/src/lib.rs (hunk 3 above)
Comment (ericmarkmartin) [reply juniper_actix_web/src/lib.rs]: Good to know—should I leave this in here for now then?

File: docs/book/content/servers/index.md (hunk 1 above)
Comment (ericmarkmartin) [reply docs/book/content/servers/index.md]: Sorry, just a mistake

File: juniper_actix_web/src/lib.rs (hunk 3 above)
Comment (ericmarkmartin) [reply juniper_actix_web/src/lib.rs]: Good to know—once I wrap this up I'd definitely be interested in taking a look

File: docs/book/content/servers/index.md (hunk 1 above)
Comment (theduke) [reply docs/book/content/servers/index.md]: No need to apologize, that's what reviews are for.

**Comment by theduke** [comment]
@ericmarkmartin thanks for working on this btw.

If you need help/advice anywhere just say so.

**Comment by ericmarkmartin** [comment]
@theduke is this PR blocked on #461? Also, is it necessary that I test the graphiql and playground functions?

**Comment by ericmarkmartin** [comment]
@theduke is there anything that needs to be fixed for this pr to be merged into the `async-await` branch?

**Comment by adwhit** [comment]
This looks really useful, but am concerned it will become immediately obsolete as appears to be based on `actix-web` v1.0? v2.0 has since been released which is wholly based off `std::future::Future`/`async/await`. Haven't looked too closely at this PR but the migrations I have done have tended to significantly simplify the code.

**Comment by ericmarkmartin** [comment]
@adwhit—I’m currently working on actix 2.0, but I’m not sure whether I should create a separate integration or not, as some integrations have taken this route while other have simply opted to replace theirs. Hence why this PR is still up.

**Comment by LegNeato** [comment]
Thanks again for the PR! I believe we landed actix 2.0 support. Do you want to still land 1.0 support? Is there anything missing from the current landed implementation that this has and should be ported there?

**Comment by ericmarkmartin** [comment]
I don't see any reason to land 1.0 given the presence of 2.0, but I will definitely see if there's anything missing from the 2.0 integration (or in its tests) that can be taken from here. Closing.
